### DISCUSSION TITLE
Permission denied (publickey)

### DISCUSSION THREAD

**Discussion by Rachel-Kenyani** [community/community#60311] [New to GitHub]
Hello everyone,
I pushed a folder and now i can't clone it into my new PC. I was thinking of just cloning one folder(Pythonprojects) from the repo using git clone -b. I configured a new SSH key made sure it was an OpenSSH public key but still the error still reoccurred. What can I do?
![image](https://github.com/community/community/assets/124130929/bbffdd1e-a227-4c2a-b77e-ed17d1011310)

**Comment by nirav-gajera** [comment]
-Try Cloning the Entire Repository: If the specific folder cloning is not working, you can try cloning the entire repository to see if it works. Use the regular git clone command without the -b flag to clone the entire repository to your new PC. Then, you can navigate to the desired folder within the cloned repository.

**Comment by h7x4** [comment]
Hello @Rachel-Kenyani.

Do you by any chance have several ssh keys?

You can debug your ssh connection to github by using the `-T` flag of ssh (this disables ssh trying to access a terminal).

```console
ssh -T git@github.com
```

To make absolutely sure that ssh is using the correct key, you can point it to the private key using the `-i` flag.

```console
ssh -T -i ~/.ssh/id_rsa git@github.com
# or
ssh -T -i ~/.ssh/id_ed25519 git@github.com
```

There's also verbosity flags to get more info: `-v` `-vv` and `-vvv`

----

If you do end up finding out that you have several keys, and you want to specify which one github should use, you can do the following:

Edit the file at `~/.ssh/config`. Add an entry that looks like this:

```
Host github.com
  User git
  IdentityFile ~/.ssh/path_to_my_private_key
```
